great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,282 @@
+    def __init__(
+        self,
+        class_name: str,
+        module_name: Optional[str] = None,
+        batch_request: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ):
+        self._class_name = class_name
+        self._module_name = module_name
+        self.batch_request = batch_request
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+            logger.info(
+                'Set "self.%s = %s" during DomainBuilderConfig instantiation', k, v
+            )
+
+    @property
+    def class_name(self) -> str:
+        return self._class_name
+
+    @property
+    def module_name(self) -> Optional[str]:
+        return self._module_name
+
+
+class DomainBuilderConfigSchema(Schema):
+    class Meta:
+        unknown = INCLUDE
+
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.domain_builder",
+    )
+    batch_request = fields.Dict(keys=fields.String(), required=False, allow_none=True)
+
+    @post_load
+    def make_config(self, data: dict, **kwargs) -> DomainBuilderConfig:
+        return DomainBuilderConfig(**data)
+
+    @post_dump
+    def remove_nulls(self, data: dict, **kwargs) -> dict:
+        res = copy.deepcopy(data)
+        for k, v in data.items():
+            if v is None:
+                res.pop(k)
+                logger.info("Removed '%s' due to null value", k)
+        return res
+
+
+class ParameterBuilderConfig(DictDot):
+    def __init__(
+        self,
+        parameter_name: str,
+        class_name: str,
+        module_name: Optional[str] = None,
+        batch_request: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ):
+        self.parameter_name = parameter_name
+        self._class_name = class_name
+        self._module_name = module_name
+        self.batch_request = batch_request
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+            logger.info(
+                'Set "self.%s = %s" during ParameterBuilderConfig instantiation', k, v
+            )
+
+    @property
+    def class_name(self) -> str:
+        return self._class_name
+
+    @property
+    def module_name(self) -> Optional[str]:
+        return self._module_name
+
+
+class ParameterBuilderConfigSchema(Schema):
+    class Meta:
+        unknown = INCLUDE
+
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.parameter_builder",
+    )
+    parameter_name = fields.String(required=True)
+    batch_request = fields.Dict(keys=fields.String(), required=False, allow_none=True)
+
+    @post_load
+    def make_config(self, data: dict, **kwargs) -> ParameterBuilderConfig:
+        return ParameterBuilderConfig(**data)
+
+    @post_dump
+    def remove_nulls(self, data: dict, **kwargs) -> dict:
```

Comment:
@cdkini Please see above idea about using `filter_properties_dict` (or the deep version thereof).  Thanks!

Reply:
I wonder if we could put this in a base class

Reply:
Moved to a base class; resolving